### PR TITLE
fix: parens before staticlookup node

### LINKED ISSUE #596: Parse Error : syntax error, unexpected '::' (T_DOUBLE_COLON), expecting ';'
Hey there!

Pretty good work. Totally amased by the prettier project!

Please see the following code sample and error. 

```php

class Countries extends RestController
{
    protected $modelClass = \App\Country::class;
    
    /**
     * Undocumented function
     *
     * @param string $key
     * @return Response
     */
    public function search($key) {
        // test auth
        if ($this->requiredAuth()) {
            return $this->respondWithUnauthenticated();
        }
        
        try {
            $data = ($this->modelClass)::where('name', 'like', strtoupper("%${key}%"))->get();
            return $this->respond($data);
        } catch (\Exception $e) {
            // TODO: Add logger ?
        }
        return $this->respondWithError('Failed retreiving data.');
    }
```

```
app\Http\Controllers\Countries.php
[error] app\Http\Controllers\Countries.php: SyntaxError: Parse Error : syntax error, unexpected '::' (T_DOUBLE_COLON), expecting ';' on line 25
[error]     at parser.raiseError (api\php\node_modules\php-parser\src\parser.js:290:17)
[error]     at parser.error (api\php\node_modules\php-parser\src\parser.js:335:15)
[error]     at parser.expectEndOfStatement (api\php\node_modules\php-parser\src\parser.js:360:10)
[error]     at parser.read_statement (api\php\node_modules\php-parser\src\parser\statement.js:383:14)
[error]     at parser.read_inner_statement (api\php\node_modules\php-parser\src\parser\statement.js:186:21)
[error]     at parser.read_inner_statements (api\php\node_modules\php-parser\src\parser\statement.js:86:30)
[error]     at parser.read_code_block (api\php\node_modules\php-parser\src\parser\statement.js:397:14)
[error]     at parser.read_statement (api\php\node_modules\php-parser\src\parser\statement.js:196:21)
[error]     at parser.read_try (api\php\node_modules\php-parser\src\parser\try.js:23:30)
[error]     at parser.read_statement (api\php\node_modules\php-parser\src\parser\statement.js:343:21)
app\Http\Controllers\RestController.php
[error] app\Http\Controllers\RestController.php: SyntaxError: Parse Error : syntax error, unexpected '::' (T_DOUBLE_COLON), expecting ';' on line 40
[error]     at parser.raiseError (api\php\node_modules\php-parser\src\parser.js:290:17)
[error]     at parser.error (api\php\node_modules\php-parser\src\parser.js:335:15)
[error]     at parser.expectEndOfStatement (api\php\node_modules\php-parser\src\parser.js:360:10)
[error]     at parser.read_statement (api\php\node_modules\php-parser\src\parser\statement.js:383:14)
[error]     at parser.read_inner_statement (api\php\node_modules\php-parser\src\parser\statement.js:186:21)
[error]     at parser.read_inner_statements (api\php\node_modules\php-parser\src\parser\statement.js:86:30)
[error]     at parser.read_code_block (api\php\node_modules\php-parser\src\parser\statement.js:397:14)
[error]     at parser.read_statement (api\php\node_modules\php-parser\src\parser\statement.js:196:21)
[error]     at parser.read_try (api\php\node_modules\php-parser\src\parser\try.js:23:30)
[error]     at parser.read_statement (api\php\node_modules\php-parser\src\parser\statement.js:343:21)
```

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #596

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   ],
   "dependencies": {
     "mem": "^3.0.1",
-    "php-parser": "3.0.0-prerelease.4"
+    "php-parser": "glayzzle/php-parser#85eb5e622fcd4ef1f70bf3b2529afe5e005bdd52"
```
Comment (evilebottnawi) [inline package.json]: Use master, for easy upgrade to stable release

File: tests/variable/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -98,8 +97,7 @@ $var = $foo->{$bar['baz']}();
 $var = $$foo['bar']['baz'];
 $var = $foo->$bar['baz'];
 $var = $foo->$bar['baz']();
-// Todo https://github.com/glayzzle/php-parser/issues/182
-// $var = (Foo::$bar)['baz']();
+$var = Foo::$bar['baz']();
```
Comment (evilebottnawi) [inline tests/variable/__snapshots__/jsfmt.spec.js.snap]: Already fixed

File: tests/variable/variable.php [hunk 3]
```diff
@@ -39,7 +39,13 @@
...
 $var = ($foo->$bar)['baz'];
 $var = ($foo->$bar)['baz']();
-// Todo https://github.com/glayzzle/php-parser/issues/182
-// $var = (Foo::$bar)['baz']();
+$var = Foo::$bar()['baz']();
+$var = (Foo::$bar())['baz']();
+$var = Foo::bar()['baz']();
+$var = (Foo::bar())['baz']();
+
+// Todo https://github.com/glayzzle/php-parser/issues/186
```
Comment (evilebottnawi) [inline tests/variable/variable.php]: Previous bug fixed, but found new bug

**Comment by evilebottnawi** [comment]
merge as is to continue work, feel free to feedback

**Comment by czosel** [comment]
:+1:

**Pull request merged**